metrics-operator/controllers/common/providers/dynatrace/dynatrace.go

```diff
@@ -95,6 +95,59 @@ func (d *KeptnDynatraceProvider) EvaluateQuery(ctx context.Context, metric metri
 	return r, b, nil
 }
 
+func (d *KeptnDynatraceProvider) EvaluateQueryForStep(ctx context.Context, metric metricsapi.KeptnMetric, provider metricsapi.KeptnMetricsProvider) ([]string, []byte, error) {
+	baseURL := d.normalizeAPIURL(provider.Spec.TargetServer)
+	query := url.QueryEscape(metric.Spec.Query)
+	var qURL string
+	if metric.Spec.Range != nil {
+		qURL = baseURL + "v2/metrics/query?metricSelector=" + query + "&from=now-" + metric.Spec.Range.Interval + "/" + metric.Spec.Range.Step
+	}
+
```

Comment:
i think we can just return an error here if `metric.Spec.Range == nil` - otherwise this function will proceed with an empty `qURL` and return an error when the `HttpClient.Do(req)` function is called

Reply:
Fixed